Shared/Samples/Create and save map/CreateAndSaveMapView.swift

```diff
@@ -0,0 +1,419 @@
+        var label: String {
+            switch self {
+            case .none:
+                "None"
+            case .timeZones:
+                "Time Zone"
+            case .census:
+                "Census"
+            }
+        }
+    }
+}
+
+private extension CreateAndSaveMapView {
+    /// The status of the create and save workflow.
+    enum Status: Equatable {
+        /// The portal is loading.
+        case loadingPortal
+        /// The portal failed to load.
+        case failedToLoadPortal
+        /// The map is being created.
+        case creatingMap
+        /// The map is being saved to the portal.
+        case savingMapToPortal
+        /// The map failed to save to the portal.
+        case failedToSaveMap
+        /// The map was saved successfully to the portal.
+        case mapSavedSuccessfully(Map)
+        /// The map is being deleted from the portal.
+        case deletingMap
+        /// The map was successfully deleted from the portal.
+        case deletedSuccessfully
+        /// The map failed to delete from the portal.
+        case failedToDelete
+    }
+}
+
+private extension CreateAndSaveMapView {
+    /// Sets up the authenticator to handle challenges.
+    func setupAuthenticator() {
+        // Setting the challenge handlers here when the model is created so user is prompted to enter
+        // credentials every time trying the sample. In real world applications, set challenge
+        // handlers at the start of the application.
+        
+        // Sets authenticator as ArcGIS and Network challenge handlers to handle authentication
+        // challenges.
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: authenticator)
+        
+        // In your application you may want to uncomment this code to persist
+        // credentials in the keychain.
+        // setupPersistentCredentialStorage()
+    }
+    
+    /// Stops the authenticator from handling the challenges and clears credentials.
+    nonisolated func teardownAuthenticator() async {
+        // Resets challenge handlers.
+        ArcGISEnvironment.authenticationManager.handleChallenges(using: nil)
+        
+        // In your application, code may need to run at a different
+        // point in time based on the workflow desired. For example, it
+        // might make sense to remove credentials when the user taps
+        // a "sign out" button.
+        await ArcGISEnvironment.authenticationManager.revokeOAuthTokens()
+        await ArcGISEnvironment.authenticationManager.clearCredentialStores()
+    }
+    
+    /// Sets up new ArcGIS and Network credential stores that will be persisted in the keychain.
+    private func setupPersistentCredentialStorage() {
+        Task {
+            try await ArcGISEnvironment.authenticationManager.setupPersistentCredentialStorage(
+                access: .whenUnlockedThisDeviceOnly,
+                synchronizesWithiCloud: false
+            )
```

Comment:
```suggestion
            try await ArcGISEnvironment.authenticationManager.setupPersistentCredentialStorage(
                access: .whenUnlockedThisDeviceOnly
            )
```

Not a big deal, but `synchronizesWithiCloud` has a default parameter.